.github/workflows/build_steps.yml

```diff
@@ -55,6 +55,13 @@ jobs:
         with:
           submodules: recursive # Just in case a dep has its own third-party deps
 
+      - name: Black Python formatting
```

Comment:
I resisted adding a step like this to avoid a trivial formatting error failing all concurrent build jobs, which wastes electricity and prevents fast feedback on other issues.

This was one of my biggest gripes with Man Build...

If you want to do this properly, please add a dedicated job to `build.py` for style checks.

Reply:
Yeah good point, or something closer to this plus https://pre-commit.com/ like Julien suggested

Reply:
IIUC, will the runs just fail early in case of misformatting in the last commits?

If this is the case, I think we can come up with a pre-commit setup that could call several hooks. This setup could include one hook for black, and potentially other ones such as another hook for clang-format. This setup could be reused on the CI allowing an identical behavior locally and on the CI and preventing maintaining several pieces of tools configuration.

What do you think?

Reply:
Yeah the `pre-commit` setup sounds good to me. Might be a bit fiddly to get working on our corporate machines, but I'm sure it will be fine.